crates/dbs-virtio-devices/src/mmio/mmio_state.rs

```diff
@@ -346,20 +346,32 @@ where
         if self.device_activated {
             warn!("reset device while it's still in active state");
         } else {
+            let mut reset_suc_flag = true;
             // . Keep interrupt_evt and queue_evts as is. There may be pending
             //   notifications in those eventfds, but nothing will happen other
             //   than supurious wakeups.
             // . Do not reset config_generation and keep it monotonically increasing
+
             for queue in self.queues.iter_mut() {
-                queue.queue = Q::new(queue.queue.max_size());
+                let new_queue = Q::new(queue.queue.max_size());
```

Comment:
This part need to be reviewed more carefully.
Other parts are just changeing the parameter names.